### PR TITLE
[SPARK-16650] Improve documentation of spark.task.maxFailures

### PR DESCRIPTION
**Pull request by tgravescs**
Clarify documentation on spark.task.maxFailures

No tests run as its documentation

### CODE REVIEW

**Comment by tgravescs** [comment]
@squito @kayousterhout

File: docs/configuration.md [hunk 1]
```diff
@@ -1188,7 +1188,9 @@ Apart from these, the following properties are also available, and may be useful
   <td><code>spark.task.maxFailures</code></td>
   <td>4</td>
   <td>
-    Number of individual task failures before giving up on the job.
+    Number of failures of any particular task before giving up on the job. The same
+    task has to fail this number of attempts, it isn't affected by the number of failures
```
Comment (kayousterhout) [inline docs/configuration.md]: for 2nd sentence maybe "Failures spread across different tasks will only cause the job to fail if one particular task has failed this number of times."?

**Comment by rxin** [comment]
I find the description "it isn't affected by the number of failures across different tasks." a little bit confusing.

**Comment by tgravescs** [comment]
@rxin  Other suggestions or should I just remove that?

**Comment by tgravescs** [comment]
oops didn't see Kay's suggestion, that sounds better to me.

File: docs/configuration.md (hunk 1 above)
Comment (tgravescs) [inline docs/configuration.md]: I can see the 2 sentence not being clear, but the second part of your suggestion about "if one particular task has failed" sounds a bit out of place to me, how about:

"The total number of failures spread across different tasks will not cause the job to fail, it has to be a particular task failing this number of attempts."

Comment (kayousterhout) [inline docs/configuration.md]: "The total number of failures spread across different tasks will not cause the job to fail; a particular task has to fail this number of attempts."? (just to make it grammatically correct)

Comment (kayousterhout) [inline docs/configuration.md]: (or your version is fine too if you don't think the grammar matters; either one is clear enough to the user I think)

Comment (tgravescs) [inline docs/configuration.md]: Sounds good. I am always open to grammar suggestions.  I will update shortly.

**Comment by squito** [comment]
lgtm

**Comment by srowen** [comment]
Merged to master/2.0

**Pull request closed without merging**